src/hotspot/share/gc/shenandoah/shenandoahNMethod.inline.hpp

```diff
@@ -28,6 +28,7 @@
 #include "gc/shared/barrierSet.hpp"
 #include "gc/shared/barrierSetNMethod.hpp"
 #include "gc/shenandoah/shenandoahNMethod.hpp"
+#include "gc/shenandoah/shenandoahClosures.inline.hpp"
```

Comment:
Is this include necessary? I cannot see why.

Reply:
I removed the same include in shenandoahUnload.cpp because it is no longer needed there. But it's needed here because of ShenandoahEvacuateUpdateMetadataClosure. It was only reachable through the include in shenandoahUnload.cpp before.

Reply:
Ah. Makes sense then.